### ISSUE TITLE
Netspeed expanding/shrinking perpetually

### ISSUE THREAD

**Issue by martywd** [mate-desktop/mate-applets#279]
#### Expected behaviour
Adjust size to maximum necessary as dictated by up/down speed displayed and then freeze position as was previous behavior in MATE Netspeed 1.16

#### Actual behaviour
Netspeed 1.18.1 constantly resizes causing icons in 'Notification Area' to jump left-to-right-to-left making mouse cursur interaction with icons in the Notification Area very difficult.

#### Steps to reproduce the behaviour
Position Netspeed applet in between Clock and Notification Area (all applets locked)

#### MATE general version
MATE 1.18.x

#### Package version
MATE Netspeed (mate-netspeed-applet) 1.18.1

#### Linux Distribution
Linux Mint MATE 18.2 Sonya 64-bit

#### Link to downstream report of your Distribution
If I'm reading this correctly, @r2rien discussed pros/cons of Netspeed constantly resizing in this previous issue post ==> https://github.com/mate-desktop/mate-applets/issues/189#issuecomment-236397650

**Comment by r2rien** [comment]
> So if you implement this, please make it optional

failed
or gtk3 switch side effect ?

**Comment by godfuture** [comment]
I think this is a good idea. I also do not like this applet changing size frequently. As this applet allows to display the unit, and the units scale over the size of 1000 why not always displaying three numbers. If the threshold changes more 1000, we could simply change the unit, but we always keep three digits. The size of the applet will not change anymore.

**Comment by lukefromdc** [comment]
I also think this should be a fixed-size applet. Jumping applets are a real nuisance, though I see from the referernced discussion that others saw this as a feature. One option would be to make fixed or variable size user-selectable, though of course the more such checks a program has to make the more time and resources (at least in theory) are needed to execute the extra code. That on the other hand is the excuse GNOME gives for removing so many options and features.

**Comment by muktupavels** [comment]
> That on the other hand is the excuse GNOME gives for removing so many options and features.

What?

**Comment by lukefromdc** [comment]
I saw some GNOME documentation a while back explaining that the more user options a program has, the more CPU cycles, RAM, etc have to be used checking them all at runtime. That was cited (I have forgotten where this was) as the reason for not trying to include all possible options and is not new.

**Comment by geckolinux** [comment]
Yep, this is definitely rather annoying. Until a more elegant solution is found would it be possible to let the user specify a fixed width for it, similar to the way the System Monitor applet works?
